chromium_src/chrome/android/java/src/org/chromium/chrome/browser/ntp/IncognitoNewTabPageView.java

```diff
@@ -0,0 +1,267 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.chromium.chrome.browser.ntp;
+
+import static org.chromium.chrome.browser.util.ViewUtils.dpToPx;
+
+import android.content.Context;
+import android.content.res.Configuration;
+import android.graphics.Canvas;
+import android.text.SpannableString;
+import android.text.method.LinkMovementMethod;
+import android.text.style.ForegroundColorSpan;
+import android.util.AttributeSet;
+import android.view.Gravity;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.annotation.IdRes;
+import androidx.annotation.StringRes;
+
+import org.chromium.base.ApiCompatibilityUtils;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.gesturenav.HistoryNavigationLayout;
+import org.chromium.chrome.browser.util.ViewUtils;
+import org.chromium.ui.text.NoUnderlineClickableSpan;
+import org.chromium.ui.text.SpanApplier;
+import org.chromium.ui.widget.ChromeBulletSpan;
+
+/**
+ * The New Tab Page for use in the incognito profile.
+ */
+public class IncognitoNewTabPageView extends HistoryNavigationLayout {
```

Comment:
Maybe we could derive our own class so we don't have to copy so much? Might just need to patch in some protected methods, but it could help simplify this I think.

Reply:
that's what we had if we patch it https://github.com/brave/brave-core/blob/8cb6f3c5c4bd09eb2cf721c8d07b18b1c52e77ff/patches/chrome-android-java-src-org-chromium-chrome-browser-ntp-IncognitoNewTabPageView.java.patch
After discussion with @bridiver we decided that the best decision is just fully replace it with our implementation.

Reply:
Ah OK, works for me then.

Reply:
yea, if we don't care about tracking changes to upstream then it's ok to copy. If we're basically abandoning the upstream implementation and going off on our own without caring what they do